### PR TITLE
Fix/prepare renderview

### PR DESCRIPTION
**Pull request by benoitdm-oslandia**
This is the 'prepare' stage to fulfil the https://github.com/qgis/QGIS-Enhancement-Proposals/issues/259 QEP.

In the next PRs we will introduce the `AbstractRenderview` concept and split the framegraph into a more modular way.

This PR does not change the framegraph logic and add some fixes to flatten the incoming changes. 

Here are the commits:

* the `qgsrenderpassquad: Do not create its own Qlayer` homogenize the layer responsibility:  it is handled outside the entity and pass to it.
* the `qgsframegraph: rework postprocessing/rendercapture nodes to enable test`: change the framegraph a bit to allow the debug map test to work!
* the `qgsframegraph: remove useless member variables for postprocessing and texture preview` remove useless member variables and make them local.
* the last commit flatten the difference between Qt5 and QT6 test results

cc @ptitjano @mkrus 

Funded by CEA/DAM @renardf

### CODE REVIEW

**Comment by benoitdm-oslandia** [comment]
gentle ping @nyalldawson @wonder-sk @lbartoletti

**Review by wonder-sk — COMMENTED** [review]
I would actually advise against doing render captures of depth maps and shadow maps. The debug depth/shadow maps are only useful while debugging shadows, they are not meant to be ever shown to normal users.

On top of that these tests will be fragile and may break often, causing maintenance burden:
- depth map's appearance depends a lot on the current setup of near/far planes, a small change in the camera's position may cause sudden big change in the depth map
- shadow rendering is IMHO quite experimental and the calculation of the shadow map texture's extent is often not good

Finally, the resulting test images also do not show much really, it is hard to tell whether the reference images are correct.

But if you want these tests mainly as a verification that nothing gets broken in the refactoring of the frame graph, then why not :slightly_smiling_face:

**Comment by nyalldawson** [comment]
No issues here from me :+1:

File: src/3d/qgsrenderpassquad.cpp [hunk 1]
```diff
@@ -43,8 +43,8 @@ typedef Qt3DCore::QGeometry Qt3DQGeometry;
 #include <QUrl>
 #include <QVector3D>
 
-QgsRenderPassQuad::QgsRenderPassQuad( QNode *parent )
-  : Qt3DCore::QEntity( parent )
+QgsRenderPassQuad::QgsRenderPassQuad( Qt3DRender::QLayer *layer, QNode *parent )
+  : Qt3DCore::QEntity()
```
[the remark this answers is not in the document]
Comment (benoitdm-oslandia) [reply src/3d/qgsrenderpassquad.cpp:47]: Just a side effect test. I will revert.

File: src/3d/qgsframegraph.cpp [hunk 2]
```diff
@@ -427,8 +429,9 @@ Qt3DRender::QFrameGraphNode *QgsFrameGraph::constructAmbientOcclusionBlurPass()
 
   mAmbientOcclusionBlurRenderCaptureTargetSelector->setTarget( depthRenderTarget );
 
-  mAmbientOcclusionBlurEntity = new QgsAmbientOcclusionBlurEntity( mAmbientOcclusionRenderTexture, mRootEntity );
-  mAmbientOcclusionBlurLayerFilter->addLayer( mAmbientOcclusionBlurEntity->layer() );
+  Qt3DRender::QLayer *ambientOcclusionBlurlayer = new Qt3DRender::QLayer();
```
[the remark this answers is not in the document]
Comment (benoitdm-oslandia) [reply src/3d/qgsframegraph.cpp:454]: I am sorry but it is the same... :)

File: src/3d/qgsframegraph.cpp [hunk 3]
```diff
@@ -279,18 +263,58 @@ Qt3DRender::QFrameGraphNode *QgsFrameGraph::constructShadowRenderPass()
...
+  mPostprocessingEntity->setObjectName( "PostProcessingPassEntity" );
+
+  return mPostProcessingCameraSelector;
+}
+
+Qt3DRender::QFrameGraphNode *QgsFrameGraph::constructSubPostPassForRenderCapture()
+{
+  Qt3DRender::QFrameGraphNode *top = new Qt3DRender::QNoDraw;
+  top->setObjectName( "Sub pass RenderCapture" );
+  top->setEnabled( true );
```
[the remark this answers is not in the document]
Comment (benoitdm-oslandia) [reply src/3d/qgsframegraph.cpp:306]: Yes indeed

File: src/3d/qgsframegraph.h [hunk 4]
```diff
@@ -344,16 +344,19 @@ class QgsFrameGraph : public Qt3DCore::QEntity
...
     Qt3DRender::QFrameGraphNode *constructRubberBandsPass();
 
+    Qt3DRender::QFrameGraphNode *constructSubPostPassForProcessing();
+    Qt3DRender::QFrameGraphNode *constructSubPostPassForRenderCapture();
+    Qt3DRender::QFrameGraphNode *constructSubPostPassForTexturesPreview();
+
     Qt3DCore::QEntity *constructDepthRenderQuad();
 
-    bool mRenderCaptureEnabled = true;
+    bool mRenderCaptureEnabled = false;
```
[the remark this answers is not in the document]
Comment (benoitdm-oslandia) [reply src/3d/qgsframegraph.h:350]: Done

File: src/3d/qgsframegraph.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (wonder-sk) [reply src/3d/qgsframegraph.cpp:454]: the upper-case L in `Layer` is the difference :wink:

**Comment by benoitdm-oslandia** [comment]
@wonder-sk 

> But if you want these tests mainly as a verification that nothing gets broken in the refactoring of the frame graph, then why not 🙂

I am totally agree with you but as you said, it is mainly to ensure non regression with the next framegraph works!

File: src/3d/qgsframegraph.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (benoitdm-oslandia) [reply src/3d/qgsframegraph.cpp:454]: okkkk

**Comment by benoitdm-oslandia** [comment]
I fixed up yours comments!

**Review by wonder-sk — APPROVED** [review]

**Comment by wonder-sk** [comment]
@benoitdm-oslandia  Thanks!

One suggestion: it is more reviewer-friendly to add new commits to the branch to address the review, rather than fixing existing commits and force-pushing them again (in the latter case, it is really hard to see changes after the review).

**Comment by benoitdm-oslandia** [comment]
> One suggestion: it is more reviewer-friendly to add new commits to the branch to address the review, rather than fixing existing commits and force-pushing them again (in the latter case, it is really hard to see changes after the review).

Yes I know but the last time it ended with multiple commit "fixup!" added to the master log because I did not have the time to squash them properly :(. It would be nice if the "fixup" commits can be automatically and properly squashed when the PR are merged!

Thanks for your approval!

**Pull request merged**